x-pack/plugins/observability_solution/entity_manager/server/lib/entities/helpers/calculate_offset.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EntityDefinition } from '@kbn/entities-schema';
+import moment from 'moment';
+import {
+  ENTITY_DEFAULT_HISTORY_FREQUENCY,
+  ENTITY_DEFAULT_HISTORY_SYNC_DELAY,
+} from '../../../../common/constants_entities';
+
+const durationToSeconds = (dateMath: string) => {
+  const parts = dateMath.match(/(\d+)([m|s|h|d])/);
+  if (!parts) {
+    throw new Error(`Invalid date math supplied: ${dateMath}`);
+  }
+  const value = parseInt(parts[1], 10);
+  const unit = parts[2] as 'm' | 's' | 'h' | 'd';
+  return moment.duration(value, unit).asSeconds();
+};
+
+export function calculateOffset(definition: EntityDefinition) {
+  const interval = durationToSeconds(definition.history.interval);
+  const syncDelay = durationToSeconds(
+    definition.history.settings.syncDelay || ENTITY_DEFAULT_HISTORY_SYNC_DELAY
+  );
+  const frequency = durationToSeconds(
+    definition.history.settings.frequency || ENTITY_DEFAULT_HISTORY_FREQUENCY
+  );
+
+  return interval + syncDelay + frequency;
```

Comment:
We should also include the latest transform frequency

> I'm tempted to add an additional minute since this calculation assumes that everything works perfectly

Agreed, what about we remote the interval but double the history frequency so this covers long-running transform but also a missed checkpoint ? We can optionally have a modifier on the frequency to cover for lags, maybe 1.2 ?

Reply:
I like the idea of doubling the frequency... for the built in the total time would become `2m + (2m * 2) = 6m` if you stick with the defaults it becomes `1m + (1m * 2) = 3m`. When I look at the freshness `(now - lastSeenTimestamp)` for the built in the max is usually `6m` and for the defaults it's around `3m` which aligns well with this approach.

I'll change it to double the `frequency` for now and if we need more delay we can add a modifier later.